kore/src/Kore/Step/Simplification/Exists.hs

```diff
@@ -156,41 +167,73 @@ simplifyEvaluated variable simplified
   | OrPattern.isTrue simplified  = return simplified
   | OrPattern.isFalse simplified = return simplified
   | otherwise = do
-    evaluated <- traverse (makeEvaluate variable) simplified
+    evaluated <- traverse (makeEvaluate [variable]) simplified
     return (OrPattern.flatten evaluated)
 
 {-| evaluates an 'Exists' given its two 'Pattern' children.
 
 See 'simplify' for detailed documentation.
 -}
 makeEvaluate
-    :: (SimplifierVariable variable, MonadSimplify simplifier)
-    => ElementVariable variable
+    :: forall simplifier variable
+    .  (SimplifierVariable variable, MonadSimplify simplifier)
+    => [ElementVariable variable]
     -> Pattern variable
     -> simplifier (OrPattern variable)
-makeEvaluate variable original
-  = fmap OrPattern.fromPatterns $ Branch.gather $ do
-    normalized <- simplifyCondition original
-    let Conditional { substitution = normalizedSubstitution } = normalized
-    case splitSubstitution variable normalizedSubstitution of
-        (Left boundTerm, freeSubstitution) ->
-            makeEvaluateBoundLeft
-                variable
-                boundTerm
-                normalized { Conditional.substitution = freeSubstitution }
-        (Right boundSubstitution, freeSubstitution) -> do
-            matched <- Monad.Trans.lift $ matchesToVariableSubstitution
-                variable
-                normalized { Conditional.substitution = boundSubstitution }
-            if matched
-                then return normalized
-                    { Conditional.predicate = Predicate.makeTruePredicate_
-                    , Conditional.substitution = freeSubstitution
-                    }
-                else makeEvaluateBoundRight
+makeEvaluate variables original = do
```

Comment:
Is the primary change here that we are simplifying multiple quantifiers at once? (The list of variables.)

Reply:
Yes, that should be it. I updated the documentation here as @traiansf suggested.